packages/remix-dev/cli/run.ts

```diff
@@ -426,13 +426,9 @@ export async function run(argv: string[] = process.argv.slice(2)) {
       break;
     }
     case "dev":
+    default: // `remix ./my-project` is shorthand for `remix dev ./my-project`
       if (!process.env.NODE_ENV) process.env.NODE_ENV = "development";
       if (flags.debug) inspector.open();
       await commands.dev(input[1], process.env.NODE_ENV);
```

Comment:
`input[1]` will return `"dev"` for `remix dev ./my-project`, no? So we need to account for that here.

---

I think it might make more sense to keep the cases separate, but factor out the common logic:

```ts
async function dev(projectDir: string) {
  if (!process.env.NODE_ENV) process.env.NODE_ENV = "development";
  if (flags.debug) inspector.open();
  await commands.dev(projectDir, process.env.NODE_ENV);
}


// ...
case "dev": { await dev(input[1]); break; }
case default: { await dev(input[0]); break; }
```

Reply:
Indeed important oversight! 👀

Reply:
Also, we should have a CLI test for the shorthand to make sure we catch this sort of thing in the future. @machour : could you add a test to that effect?

Reply:
glad you caught that @pcattori 😱 I'll make the requested change and test

Reply:
I definitely need some guidance for the test part, as the `dev` command itself is not currently tested, and I'm a bit clueless about how it should be done.

Reply:
I could test that both `remix non-existing-dir` and `remix dev non-existing-dir` print an error matching `Error loading Remix config in .*/non-existing-dir/remix.config.js` confirming that they both really do the same thing with args, but that seem a bit flaky to me 😅